pkg/services/service.go

```diff
@@ -79,6 +86,16 @@ func (e *Engine) GoTick(ticker *timeutil.Ticker, fn func(context.Context)) {
 	})
 }
 
+// TraceAs wraps fn in a tracing span with service attributes and the given name.
+// Consider wrapping funcs passed to Go or GoTick.
+func (e *Engine) TraceAs(name string, fn func(context.Context)) func(context.Context) {
```

Comment:
would it be helpful to add a variadic strings param for key value pairs to associate with the trace?

Reply:
I think I would rather just expose the Tracer than make this complex. Any kind of variadic after the func can be awkward since these are often large closures.

Reply:
Update to `Tracer() trace.Tracer` and added an example